internal/website/content/howto/blob/_index.md

```diff
@@ -9,3 +9,246 @@ providers and accessing them via HTTP. These guides show how to work with
+store and access data from it using the standard Go I/O patterns.
+
+[`blob.OpenBucket`]:
+https://godoc.org/gocloud.dev/blob#OpenBucket
+["blank import"]: https://golang.org/doc/effective_go.html#blank_import
+[Concepts: URLs]: {{< ref "/concepts/urls.md" >}}
+
+### Prefixed Buckets {#prefix}
+
+You can wrap a `*blob.Bucket` to always operate on a subfolder of the bucket
+using `blob.PrefixedBucket`:
+
+{{< goexample "gocloud.dev/blob.ExamplePrefixedBucket" >}}
+
+Alternatively, you can configure the prefix directly in the `blob.OpenBucket`
+URL:
+
+{{< goexample "gocloud.dev/blob.Example_openFromURLWithPrefix" >}}
+
+## Writing Data to a Bucket {#writing}
+
+To write data to a bucket, you create a writer, write data to it, and then
+close the writer. Closing the writer commits the write to the provider,
+flushing any buffers, and releases any resources used while writing, so you
+must always check the error of `Close`.
+
+The writer implements [`io.Writer`][], so you can use any functions that take
+an `io.Writer` like `io.Copy` or `fmt.Fprintln`.
+
+{{< goexample src="gocloud.dev/blob.ExampleBucket_NewWriter" imports="0" >}}
+
+In some cases, you may want to cancel an in-progress write to avoid the blob
+being created or overwritten. A typical reason for wanting to cancel a write
+is encountering an error in the stream your program is copying from. To abort
+a write, you cancel the `Context` you pass to the writer. Again, you must
+always `Close` the writer to release the resources, but in this case you can
+ignore the error because the write's failure is expected.
+
+{{< goexample src="gocloud.dev/blob.ExampleBucket_NewWriter_cancel" imports="0" >}}
+
+[`io.Writer`]: https://golang.org/pkg/io/#Writer
+
+## Reading Data from a Bucket {#reading}
+
+Once you have written data to a bucket, you can read it back by creating a
+reader. The reader implements [`io.Reader`][], so you can use any functions
+that take an `io.Reader` like `io.Copy` or `io/ioutil.ReadAll`. You must
+always close a reader after using it to avoid leaking resources.
+
+{{< goexample src="gocloud.dev/blob.ExampleBucket_NewReader" imports="0" >}}
+
+Many storage providers provide efficient random-access to data in buckets. To
+start reading from an arbitrary offset in the blob, use `NewRangeReader`.
+
+{{< goexample src="gocloud.dev/blob.ExampleBucket_NewRangeReader" imports="0" >}}
+
+[`io.Reader`]: https://golang.org/pkg/io/#Reader
+
+## Deleting Blobs {#deleting}
+
+You can delete blobs using the `Bucket.Delete` method.
+
+{{< goexample src="gocloud.dev/blob.ExampleBucket_Delete" imports="0" >}}
+
+## Other Operations {#other}
+
+These are the most common operations you will need to use with a bucket.
+Other operations like listing and reading metadata are documented in the
+[`blob` package documentation][].
+
+[`blob` package documentation]: https://godoc.org/gocloud.dev/blob
+
+## Service-Specific Initialization
+
+### S3 {#s3}
```

Comment:
While you're here, how about fixing http://b/136690923 to update the order?

Reply:
Done